pkg/plugins/runtime/k8s/webhooks/injector/injector.go

```diff
@@ -215,7 +218,7 @@ func (i *KumaInjector) InjectKuma(ctx context.Context, pod *kube_core.Pod) error
 
 	pod.Spec.InitContainers = append(append(prependInitContainers, pod.Spec.InitContainers...), appendInitContainers...)
 
-	if err := i.overrideHTTPProbes(pod); err != nil {
+	if err := probes.OverridePodProbes(pod, log); err != nil {
```

Comment:
I think the behaviour here should be safer.

It should be:

1. virtualProbesEnabled by default
2. if virtualProbesEnabled is disabled fallback to the old way

This enables users to fallback to what was done before if switching to this new option doesn't work as expected.